socket.c

```diff
@@ -533,11 +535,25 @@ static int ensure_tls_handshake(camblet_socket *s, struct msghdr *msg)
 			pr_debug("TLS handshake done # command[%s] sk[%p]", current->comm, s->sock);
 			trace_msg(conn_ctx, "TLS handshake done", 0);
 		}
+		else if (ret == -ERESTARTSYS)
+		{
+			pr_debug("TLS handshake got interrupted by signal # command[%s] sk[%p]", current->comm, s->sock);
+			ret = -EINTR;
+			goto bail;
+		}
+		else if (s->sock->sk_err || s->sock->sk_state == TCP_CLOSE ||
+				 (s->sock->sk_shutdown & RCV_SHUTDOWN))
```

Comment:
It might be worthwhile to refactor this "is closed check" into a nice little method..